datarobot_batch_scoring/network/network.py

```diff
@@ -28,6 +28,40 @@
 logger = logging.getLogger(__name__)
 FakeResponse = collections.namedtuple('FakeResponse', 'status_code, text')
 
+from requests.packages.urllib3.util.connection import socket as r_socket
+try:
+    from functools import lru_cache
+except ImportError:
+    # https://stackoverflow.com/questions/17119154/python-decorator-optional-argument
+    import functools
+    def lru_cache(*setting_args, **setting_kwargs):
+        cache = {}
+        no_args = (
+            len(setting_args) == 1 and
+            not setting_kwargs and
+            callable(setting_args[0])
+        )
+        if no_args:
+            # We were called without args
+            func = setting_args[0]
+        def outer(func):
+            @functools.wraps(func)
+            def cacher(*args, **kwargs):
+                key = tuple(args) + tuple(sorted(kwargs.items()))
+                if key not in cache:
```

Comment:
There is no locking in this implementation, so it can still have threading issues :)

Reply:
> There is no locking in this implementation
ay ay ay. Believe me, the code is pretty crowded as is. 

Also, there is only one address resolved, to my knowledge, so it's just a read -- I don't believe you can get an error without conflicting writes.